src/dstack/api/server/_files.py

```diff
@@ -0,0 +1,18 @@
+from typing import BinaryIO
+
+from pydantic import parse_obj_as
+
+from dstack._internal.core.models.files import FileArchive
+from dstack._internal.server.schemas.files import GetFileArchiveByHashRequest
+from dstack.api.server._group import APIClientGroup
+
+
+class FilesAPIClient(APIClientGroup):
+    def get_archive_by_hash(self, hash: str) -> FileArchive:
```

Comment:
Any plans to use this function?

Reply:
`dstack` itself does not use this API method at the moment, but it can be used to avoid uploading a large archive if it's already uploaded (e.g., in `RunCollection.apply_plan`):

1. Create an archive, get a hash
2. Check if the archive is already uploaded, calling this method
3. If the archive is already uploaded, reuse `id` returned by the method, upload the archive otherwise

We could improve `RunCollection.apply_plan` by adding this additional check for large archives (for small ones the additional API call only increases execution time).